torch/csrc/api/src/nn/modules/pooling.cpp

```diff
@@ -273,6 +273,91 @@ template class MaxUnpoolImpl<3, MaxUnpool3dImpl>;
 
 // ============================================================================
 
+FractionalMaxPool2dImpl::FractionalMaxPool2dImpl(const FractionalMaxPool2dOptions& options_)
+    : options(options_) {
+  reset();
+}
+
+void FractionalMaxPool2dImpl::reset() {
+  _random_samples = register_buffer("_random_samples", options._random_samples());
+  if (options.output_size() == c10::nullopt && options.output_ratio() == c10::nullopt) {
+    TORCH_CHECK(
+      false,
+      "FractionalMaxPool2d requires specifying either ",
+      "an output size, or a pooling ratio");
+  }
+  if (options.output_size() != c10::nullopt && options.output_ratio() != c10::nullopt) {
+    TORCH_CHECK(false, "only one of output_size and output_ratio may be specified");
+  }
+  if (options.output_ratio() != c10::nullopt) {
+    at::IntArrayRef output_ratio = at::IntArrayRef(options.output_ratio().value());
+    if (!(0 < output_ratio[0] && output_ratio[0] < 1 &&
+          0 < output_ratio[1] && output_ratio[1] < 1)) {
+      TORCH_CHECK(false, "output_ratio must be between 0 and 1 (got ", output_ratio, ")");
+    }           
+  }
+}
+
+Tensor FractionalMaxPool2dImpl::forward(const Tensor& input) {
+  return F::detail::fractional_max_pool2d(
+            input, options.kernel_size(), options.output_size(), options.output_ratio(),
+            _random_samples=_random_samples);
```

Comment:
I didn't get it, maybe just
```suggestion
            _random_samples);
```

Reply:
I was surprised this doesn't throw a compile error :/ Fixed.